### PR TITLE
Support for different types of drilldowns and drilldowns with filters

### PR DESCRIPTION
**Pull request by artkzlv**
**Url example for drilldown with filters:** 
`screen/screenName/view/viewName?filters={"someBcName":"name.contains=11%26status.equalsOneOf=[\\"New\\", \\"Fixed\\"]", "someBcName2": "someField.contains=value"}`
Url parameter "filters" is valid JSON object with bcNames and filters. 
Responsibility for escaping special characters ("%26", "\", etc.) in JSON on whoever writes JSON.

### CODE REVIEW

File: src/epics/router.ts [hunk 1]
```diff
@@ -188,6 +229,53 @@ const handleRouter: Epic = (action$, store) => action$.ofType(types.handleRouter
...
+                fieldName,
+                type: type as FilterType,
+                value
+            })
+        }
+    })
+    return result.length ? result : null
+}
+
+function isManyValueFilterType(type: string) {
```
Comment (Dergash) [inline src/epics/router.ts]: It's possible to inline that check as it's pretty basic and not reused anywhere

File: src/epics/router.ts [hunk 2]
```diff
@@ -188,6 +229,53 @@ const handleRouter: Epic = (action$, store) => action$.ofType(types.handleRouter
...
+    return (
+        type === FilterType.containsOneOf ||
+        type === FilterType.equalsOneOf
+    )
+}
+
+function locationChanged(pathname: string, currentScreenName: string, currentViewName: string)  {
```
Comment (Dergash) [inline src/epics/router.ts]: [There is a function](https://github.com/tesler-platform/tesler-ui/blob/a0ac06237d57cfa359fdb26ffb7954b8b56b62e9/src/Provider.tsx#L123) that is used for splitting into the `Route` object. This function could be overridden from client side as long as contract have been followed, i.e. it does not necessary will work with this specific splitting algorithm.

`locationChanged` should be called with two `Route` arguments instead

File: src/epics/router.ts [hunk 3]
```diff
@@ -130,12 +133,47 @@ const changeView: Epic = (action$, store) => action$.ofType(types.selectView)
...
-    // const state = store.getState() as Store
+    const state = store.getState() as Store
     const url = action.payload.url
     switch (action.payload.drillDownType) {
-        case 'inner':
+        case DrillDownType.external:
+            window.location.href = url
+            return Observable.empty()
+        case DrillDownType.externalNew:
+            if (/^[a-z0-9]+:\/\//i.test(url)) {
```
Comment (Dergash) [inline src/epics/router.ts]: Will it work for urls without protocol specified, e.g. "www.github.com"?

File: src/epics/router.ts [hunk 4]
```diff
@@ -130,12 +133,47 @@ const changeView: Epic = (action$, store) => action$.ofType(types.selectView)
 
 const drillDown: Epic = (action$, store) => action$.ofType(types.drillDown)
 .switchMap(action => {
-    // const state = store.getState() as Store
+    const state = store.getState() as Store
```
Comment (Dergash) [inline src/epics/router.ts]: `as Store` is not required anymore as `store.getState` return value now has a correct typing

File: src/epics/router.ts [hunk 5]
```diff
@@ -130,12 +133,47 @@ const changeView: Epic = (action$, store) => action$.ofType(types.selectView)
...
     const url = action.payload.url
     switch (action.payload.drillDownType) {
-        case 'inner':
+        case DrillDownType.external:
+            window.location.href = url
+            return Observable.empty()
```
Comment (Dergash) [inline src/epics/router.ts]: There is a bit of inconsistency with `return Observable.empty()` and `break` operators. We end up at `return Observable.empty()` anyway (line 179) so it probably better to just use `break` for case branches

File: src/epics/router.ts (hunk 3 above)
Comment (artkzlv) [reply src/epics/router.ts]: "window.open" always opens relative url if it came without a protocol

File: src/epics/router.ts (hunk 1 above)
Comment (artkzlv) [reply src/epics/router.ts]: fixed

File: src/epics/router.ts (hunk 2 above)
Comment (artkzlv) [reply src/epics/router.ts]: Fixed, function locationChanged removed

File: src/epics/router.ts (hunk 4 above)
Comment (artkzlv) [reply src/epics/router.ts]: Fixed

File: src/epics/router.ts (hunk 5 above)
Comment (artkzlv) [reply src/epics/router.ts]: Fixed

**Pull request merged**